### ISSUE TITLE
Add a feature to add N braces for the number of arguments when completing a command

### ISSUE THREAD

**Issue by juliankrieger** [neoclide/coc-vimtex#4]
Latex commands are defined like this
`\newcommand{\newcommandname}[Number of Arguments]
`

Is it possible to add a variable number of braces when completing a command, for example make
`\begin` complete to `\begin{}`, and make the braces "jumpable" like with snippet plugins?

I would be willing to look into it, but I'd love a suggestion on where to start.

**Comment by chemzqm** [comment]
Yes, you just need to generate snippet according to current user input, just like what https://github.com/neoclide/coc-emmet does.

**Comment by p135246** [comment]
Hi, are there any news? I would be also very much interested in this sort of auto-expansion. Unfortunately, I do not know the internal mechanisms of coc and vimtex and can not do it other than by defining a new snippet for every LaTeX command...
